Fix cookies sending by ResponseEmitter.

ADmad opened the pull request:
Before this fix the cookies info in `Response::$_cookies` was never used to actually set cookies.

I haven't added/updated tests as I am not 100% sure this is the correct way to fix this problem but testing in an actual app it does seem to fix the problem.

ADmad committed:
Fix cookies sending by ResponseEmitter.

antograssiot commented on src/Http/ResponseEmitter.php:
```diff
@@ -162,9 +162,11 @@ protected function emitStatusLine(ResponseInterface $response)
      */
     protected function emitHeaders(ResponseInterface $response)
     {
+        $cookies = $response->cookie();
```
Shouldn't it be `getCookie()` here ?

ADmad replied:
Nope because `getCookie()` requires cookie name and doesn't return all cookies.

antograssiot replied:
Then we should maybe check the response class before calling `cookie()` method to avoid the errors shown by the CI

antograssiot commented:
solved it in my app also

markstory commented on src/Http/ResponseEmitter.php:
```diff
@@ -162,9 +162,14 @@ protected function emitStatusLine(ResponseInterface $response)
     {
+        $cookies = [];
+        if (method_exists($response, 'cookie')) {
+            $cookies = $response->cookie();
```
I wonder if we should add a getCookies() method.

markstory replied:
I added the `getCookies()` method and added tests.

markstory reviewed (commented):
I can add  a test before merging this.

markstory committed:
Add getCookies() method to response.

this method better follows the segregated getter method style we've been
adopting.

markstory committed:
Fix reason phrase not getting set by statusCode()

Setting a statuscode with the mutable interface should set the reason
phrase.

markstory committed:
Use getCookies() and add test for emitting cookies

antograssiot commented:
`Cake\Test\TestCase\Network\ResponseTest::testGetReasonPhrase`also needs to be updated if the default phrase is `'OK'` instead of null

markstory commented:
I will get that fixed.

markstory committed:
Fix reasonPhrase having a default value that matches the status code.
